dev/release/02-source.sh

```diff
@@ -67,11 +68,23 @@ cd -
 
 # build source archive for Apache Arrow GLib by "make dist".
 cd ${extract_dir}/c_glib
+
+if [ $(uname) = "Darwin" ]; then
+  export PKG_CONFIG_PATH=$PKG_CONFIG_PATH:$(brew --prefix libffi)/lib/pkgconfig
+  export XML_CATALOG_FILES=/usr/local/etc/xml/catalog
+fi
+
+export PKG_CONFIG_PATH=$PKG_CONFIG_PATH:$cpp_install_dir/lib/pkgconfig
+export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$cpp_install_dir/lib
```

Comment:
We need to use `$cpp_install_dir/lib` path than pre-defined paths. Because if we have installed Arrow in pre-defined paths, it's used than the built Arrow in this shell script.

How about the following style?

```shell
if [ $(uname) = "Darwin" ]; then
  export PKG_CONFIG_PATH=$(brew --prefix libffi)/lib/pkgconfig:$PKG_CONFIG_PATH
  export XML_CATALOG_FILES=/usr/local/etc/xml/catalog
fi

export PKG_CONFIG_PATH=$cpp_install_dir/lib/pkgconfig:$PKG_CONFIG_PATH
export LD_LIBRARY_PATH=$cpp_install_dir/lib:$LD_LIBRARY_PATH
```